src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -55,9 +56,43 @@
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WritePackageUsageData" />
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WriteUsageReports" />
 
-  <!-- Update NuGet feeds in the repo -->
+  <Target Name="CopyNuGetConfig"
+          BeforeTargets="
+            UpdateNuGetConfig;
+            Build"
+          Condition="'$(NuGetConfigFile)' != '' or '@(NuGetConfigFiles)' != ''">
+
+    <!-- Update the detected or manually specified NuGetConfigFile, but also allow multiple. -->
+    <ItemGroup>
+      <NuGetConfigFiles Include="$(NuGetConfigFile)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <ModifiedNugetConfigFilesIntermediateRoot>$(IntermediateOutputPath)nuget.config/</ModifiedNugetConfigFilesIntermediateRoot>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <UpdatedNuGetConfigFiles Include="@(NuGetConfigFiles->'%(FullPath)'->Replace($(RepoRoot), $(ModifiedNugetConfigFilesIntermediateRoot)))" Condition="'@(NuGetConfigFiles)' != ''">
+        <OriginalPath>%(FullPath)</OriginalPath>
+      </UpdatedNuGetConfigFiles>
+    </ItemGroup>
+
+    <MakeDir Directories="@(UpdatedNuGetConfigFiles -> '%(RootDir)%(Directory)')" />
+
+    <Copy
+      SourceFiles="@(UpdatedNuGetConfigFiles -> '%(OriginalPath)')"
+      DestinationFiles="@(UpdatedNuGetConfigFiles -> '%(FullPath)')" />
+
+    <ItemGroup>
+      <NuGetConfigFiles Remove="@(NuGetConfigFiles)" />
+      <NuGetConfigFiles Include="@(UpdatedNuGetConfigFiles)" />
+    </ItemGroup>
+
+  </Target>
+
   <Target Name="UpdateNuGetConfig"
           BeforeTargets="Build"
+          AfterTargets="CopyNuGetConfig"
```

Comment:
If you make this a `DependsOnTargets` here you can remove L60-62 (the BeforeTargets).